### PR TITLE
Geometry, Constants, Chai

### PR DESCRIPTION
**Pull request by cixzhang**
Adding geometry, constants, and using Chai for assertion handling.

### CODE REVIEW

File: geometry.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+function segmentIntersect (a, vA, b, b2) {
+  var vB = v3.subtract(b2, b),
+      i = lineIntersect(a, vA, b, vB);
+
+  if (!i) return i;
+
+  var vI = v3.subtract(i, b),
+      projection = v3.dot(vI, vB);
+
+  if (projection <= 1 && projection >= 0) return i;
```
Comment (ptpham) [inline geometry.js]: It's unclear to me that this handles a b vector of length greater than 1 correctly. Could you add a test case where the line segment is intersected and the the length of b is greater than 1?

File: test/geometry.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
+
+var assert = require('chai').assert,
+    EPSILON = require('../constants.js').EPSILON,
+    v3 = require('../twgl.js').v3,
+    Geometry = require('../geometry.js');
```
Comment (ptpham) [inline test/geometry.js]: Lowercase geometry for consistency? =)

File: geometry.js (hunk 1 above)
Comment (cixzhang) [inline geometry.js]: Yeah, forgot to do a division @_@

**Comment by ptpham** [comment]
Sorry about the Geometry capitalization comment, I was confused. I've been working with a different convention for work. XD Feel free to change it back without a pull request. Good work!

**Comment by ptpham** [comment]
Oh, oh, actually, we need to discuss the capitalization. XD At the moment I'm thinking lowercase for a collection of functions and uppercase for a class. =P

**Comment by cixzhang** [comment]
Sure, I kept it lowercase anyway. :P What about a collection of constants? Lowercase as well?

**Comment by ptpham** [comment]
I think that makes sense. =D =D =D =D

**Pull request merged**